src/combinatorics/quiver/symmetric.lean

```diff
@@ -0,0 +1,214 @@
+  rw reverse_eq_reverse_iff,
+end
+
+variables {U V W}
+
+/-- A prefunctor preserving reversal of arrows -/
+class _root_.prefunctor.map_reverse [has_reverse U] [has_reverse V] (φ : U ⥤q V) :=
+(map_reverse' : ∀ {u v : U} (e : u ⟶ v), φ.map (reverse e) = reverse (φ.map e))
+
+@[simp] lemma _root_.prefunctor.map_reverse'  [has_reverse U] [has_reverse V] (φ : U ⥤q V)
+  [φ.map_reverse] {u v : U} (e : u ⟶ v) : φ.map (reverse e) = reverse (φ.map e) :=
+prefunctor.map_reverse.map_reverse' e
+
+instance _root_.prefunctor.map_reverse_comp [quiver.{w+1} W]
+  [has_reverse U] [has_reverse V] [has_reverse W] (φ : U ⥤q V) (ψ : V ⥤q W)
+  [φ.map_reverse] [ψ.map_reverse] : (φ ⋙q ψ).map_reverse :=
+{ map_reverse' := λ u v e, by { simp only [prefunctor.comp_map, prefunctor.map_reverse'], } }
+
+instance _root_.prefunctor.map_reverse_id [has_reverse U] :
+  (prefunctor.id U).map_reverse :=
+{ map_reverse' := λ u v e, rfl }
+
+instance : has_reverse (symmetrify V) := ⟨λ a b e, e.swap⟩
+instance : has_involutive_reverse (symmetrify V) :=
+{ to_has_reverse := ⟨λ a b e, e.swap⟩,
+  inv' := λ a b e, congr_fun sum.swap_swap_eq e }
+
+@[simp] lemma symmetrify_reverse {a b : symmetrify V} (e : a ⟶ b) :
+  reverse e = e.swap := rfl
+
+/-- Shorthand for the "forward" arrow corresponding to `f` in `symmetrify V` -/
+abbreviation hom.to_pos {X Y : V} (f : X ⟶ Y) :
+  (quiver.symmetrify_quiver V).hom X Y := sum.inl f
+
+/-- Shorthand for the "backward" arrow corresponding to `f` in `symmetrify V` -/
+abbreviation hom.to_neg {X Y : V} (f : X ⟶ Y) :
+  (quiver.symmetrify_quiver V).hom Y X := sum.inr f
+
+/-- Reverse the direction of a path. -/
+@[simp] def path.reverse [has_reverse V] {a : V} : Π {b}, path a b → path b a
+| a path.nil := path.nil
+| b (path.cons p e) := (reverse e).to_path.comp p.reverse
+
+@[simp] lemma path.reverse_to_path [has_reverse V] {a b : V} (f : a ⟶ b) :
+  f.to_path.reverse = (reverse f).to_path := rfl
+
+@[simp] lemma path.reverse_comp [has_reverse V] {a b c : V} (p : path a b) (q : path b c) :
+  (p.comp q).reverse = q.reverse.comp p.reverse := by
+{ induction q, { simp, }, { simp [q_ih], }, }
+
+@[simp] lemma path.reverse_reverse [h : has_involutive_reverse V] {a b : V} (p : path a b) :
+  p.reverse.reverse = p :=
+begin
+  induction p,
+  { simp, },
+  { simp only [path.reverse, path.reverse_comp, path.reverse_to_path, reverse_reverse, p_ih],
+    refl, },
+end
+
+namespace symmetrify
+
+/-- The inclusion of a quiver in its symmetrification -/
+def of : prefunctor V (symmetrify V) :=
+{ obj := id,
+  map := λ X Y f, sum.inl f }
+
+/-- Given a quiver `V'` with reversible arrows, a prefunctor to `V'` can be lifted to one from
+    `symmetrify V` to `V'` -/
+def lift {V' : Type*} [quiver V'] [has_reverse V'] (φ : V ⥤q V') :
+  (symmetrify V) ⥤q V' :=
+{ obj := φ.obj,
+  map := λ X Y f, sum.rec (λ fwd, φ.map fwd) (λ bwd, reverse (φ.map bwd)) f }
+
+lemma lift_spec  (V' : Type*) [quiver V'] [has_reverse V'] (φ : V ⥤q V') :
```

Comment:
Can you avoid repeating those variables everywhere?

Reply:
I get universe errors if I make `V', quiver V', has_reverse V'` into variables… let me try and debug it.

Reply:
Try universe-ascripting. If you get errors when putting them to a `variables` declaration, it's probably because `V'` must be in the same universe as `V`.

Reply:
mmh, isn't it better to leave as much free as possible? Isn't this kind of problem a symptom of something too restrictive up the river?

Reply:
It's not any more general now. Having `V' : Type*` means that Lean is free to unify the universes if it needs. Do `set
_option pp.universes true` and have a look for yourself.

Reply:
I think I've managed without too much trouble. Do tell me if that's good for you? I'll need to update the mathlib4 PR too, but only once this one is merged if that's OK.